org.gecko.emf.exporter/src/org/gecko/emf/exporter/AbstractEMFExporter.java

```diff
@@ -0,0 +1,1957 @@
+		return ((boolean) exportOptions.getOrDefault(EMFExportOptions.OPTION_EXPORT_NONCONTAINMENT, Boolean.FALSE));
+	}
+
+	protected boolean exportMetadataEnabled(Map<Object, Object> exportOptions) {
+		return ((boolean) exportOptions.getOrDefault(EMFExportOptions.OPTION_EXPORT_METADATA, Boolean.TRUE));
+	}
+
+	protected boolean addMappingTableEnabled(Map<Object, Object> exportOptions) {
+		return ((boolean) exportOptions.getOrDefault(EMFExportOptions.OPTION_ADD_MAPPING_TABLE, Boolean.FALSE));
+	}
+
+	protected boolean showURIsEnabled(Map<Object, Object> exportOptions) {
+		return ((boolean) exportOptions.getOrDefault(EMFExportOptions.OPTION_SHOW_URIS, Boolean.TRUE));
+	}
+
+	protected boolean showREFsEnabled(Map<Object, Object> exportOptions) {
+		return ((boolean) exportOptions.getOrDefault(EMFExportOptions.OPTION_SHOW_REFS, Boolean.TRUE));
+	}
+
+	protected void resetStopwatch() {
+		this.stopwatch.reset().start();
+	}
+
+	protected double elapsedTimeInSeconds() {
+		return (double) this.stopwatch.elapsed().toNanos() / 1000000000.0;
+	}
+
+	protected class ProcessedEObjectsDTO {
+		// maps matrices' names to matrices
+		public final Map<String, Table<Integer, Integer, Object>> matrixNameToMatrixMap;
+
+		// maps EObjects' IDs to names of matrices, so those can be looked up e.g. when
+		// constructing links
+		public final Map<String, String> eObjectIDToMatrixNameMap;
+
+		// maps EObjects' unique identifiers to pseudo IDs - for those EObjects which
+		// lack id field
+		public final Map<String, String> eObjectUniqueIdentifierToPseudoIDMap;
+
+		// stores EObjects' EClasses for which pseudo IDs where generated
+		public final Set<EClass> eObjectsClassesWithPseudoIDs;
+
+		// stores EObjects' EClasses - used e.g. to construct meta data
+		public final Set<EClass> eObjectsClasses;
+
+		// stores EEnums - used e.g. to construct meta data
+		public final Set<EEnum> eObjectsEnums;
+
+		// lookup index for storing information about position (row number) of specific
+		// reference
+		public final Map<EMFExportRefMatrixNameIDCompositeKey, Integer> refMatrixRowKeyIndex;
+
+		public ProcessedEObjectsDTO() {
+			this.matrixNameToMatrixMap = new HashMap<String, Table<Integer, Integer, Object>>();
+			this.eObjectIDToMatrixNameMap = new HashMap<>();
+			this.eObjectUniqueIdentifierToPseudoIDMap = new HashMap<>();
+			this.eObjectsClassesWithPseudoIDs = new HashSet<>();
+			this.eObjectsClasses = new HashSet<>();
+			this.eObjectsEnums = new HashSet<>();
+			this.refMatrixRowKeyIndex = new HashMap<>();
+		}
+	}
+
+	private List<EObject> preProcessResourceEObjects(Resource resource, Map<Object, Object> exportOptions) {
+		resetStopwatch();
+
+		logger.info("Starting preprocessing EObjects contained in Resource");
+
+		final PreProcessedResourceEObjectsDTO preProcessedEObjectsDTO = new PreProcessedResourceEObjectsDTO();
+
+		preProcessedEObjectsDTO.rawResourceEObjects.addAll(resource.getContents());
+
+		preProcessedEObjectsDTO.resourceEObjectsURIs
```

Comment:
You should store the resource as well, because then you can simply compare `eResource()` of an EObject (if present) with the given resource without doing complex URI matching.